test/src/module/config/parseTest.c

```diff
@@ -536,84 +536,11 @@ testRun(void)
-        TEST_RESULT_DOUBLE(valueDbl, 10, "valueDbl B to bytes");
-        TEST_RESULT_STR_Z(value, "10", "value B to bytes");
-
-        strCatZ(value, "Kb");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 10240, "valueDbl KB to bytes");
-        TEST_RESULT_STR_Z(value, "10240", "value KB to bytes");
-
-        strTrunc(value, strChr(value, '2'));
-        strCatZ(value, "k");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 10240, "valueDbl k to bytes");
-        TEST_RESULT_STR_Z(value, "10240", "value k to bytes");
-
-        strCatZ(value, "pB");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 11529215046068469760U, "valueDbl Pb to bytes");
-        TEST_RESULT_STR_Z(value, "11529215046068469760", "value Pb to bytes");
-
-        strTrunc(value, strChr(value, '5'));
-        strCatZ(value, "GB");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 11811160064U, "valueDbl GB to bytes");
-        TEST_RESULT_STR_Z(value, "11811160064", "value GB to bytes");
-
-        strTrunc(value, strChr(value, '8'));
-        strCatZ(value, "g");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 11811160064U, "valueDbl g to bytes");
-        TEST_RESULT_STR_Z(value, "11811160064", "value g to bytes");
-
-        strTrunc(value, strChr(value, '8'));
-        strCatZ(value, "T");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 12094627905536U, "valueDbl T to bytes");
-        TEST_RESULT_STR_Z(value, "12094627905536", "value T to bytes");
-
-        strTrunc(value, strChr(value, '0'));
-        strCatZ(value, "tb");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 13194139533312U, "valueDbl tb to bytes");
-        TEST_RESULT_STR_Z(value, "13194139533312", "value tb to bytes");
-
-        strTrunc(value, strChr(value, '3'));
-        strCatZ(value, "0m");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 10485760, "valueDbl m to bytes");
-        TEST_RESULT_STR_Z(value, "10485760", "value m to bytes");
-
-        strCatZ(value, "mb");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_DOUBLE(valueDbl, 10995116277760U, "valueDbl mb to bytes");
-        TEST_RESULT_STR_Z(value, "10995116277760", "value mb to bytes");
-
-        strTrunc(value, strChr(value, '0'));
-        strCatZ(value, "99999999999999999999p");
-        convertToByte(&value, &valueDbl);
-        TEST_RESULT_STR_Z(value, "225179981368524800000000000000000000", "value really large  to bytes");
+        TEST_RESULT_UINT(convertToByte(STRDEF("10B")), 10, "10B");
+        TEST_RESULT_UINT(convertToByte(STRDEF("1k")), 1024, "1k");
+        TEST_RESULT_UINT(convertToByte(STRDEF("5G")), (uint64_t)5 * 1024 * 1024 * 1024, "5G");
+        TEST_RESULT_UINT(convertToByte(STRDEF("3Tb")), (uint64_t)3 * 1024 * 1024 * 1024 * 1024, "3Tb");
```

Comment:
I'd like to see the following added which will test no qualifier, megabytes and our max PB so that we cover all the qualifiers we accept and our max range:
```
TEST_RESULT_UINT(convertToByte(STRDEF("11")), 11, "11 - no qualifier, default bytes");
TEST_RESULT_UINT(convertToByte(STRDEF("4pB")), 4503599627370496, "4pB");
TEST_RESULT_UINT(convertToByte(STRDEF("15MB")), (uint64_t)15 * 1024 * 1024, "15MB");
```

Reply:
Added in 6205110cd0b05.